bundles/framework/myplaces3/view/PlaceForm.js

```diff
@@ -301,5 +286,176 @@ Oskari.clazz.define('Oskari.mapframework.bundle.myplaces3.view.PlaceForm',
         _getOnScreenForm: function () {
             // unbind live so
             return jQuery('div.myplacesform').filter(':visible');
+        },
+        /**
+         * @method _initializePlace
+         * Initializes place to ensure that we don't have null pointers
+         * @private
+         */
+        _initializePlace: function () {
+            if (!this.place) {
+                this.place = Oskari.clazz.create('Oskari.mapframework.bundle.myplaces3.model.MyPlace');
+                const initialCategory = this.options && typeof this.options.category !== 'undefined' ? this.options.category : 1;
+                this.place.setCategoryId(initialCategory);
+            }
+        },
+        createEditDialog: function () {
+            this._initializePlace(); // initialize place so we have empty place to fill on
+            this._populateForm(); // populate form with data from place
+
+            this.dialog = Oskari.clazz.create('Oskari.userinterface.component.Popup'); // Create popup dialog
+            this.dialog.makeDraggable();
+
+            // add new dialog to ui
+            this.dialog.show(this.loc('placeform.title'), '<div class="places-edit-dialog"></div>');
+
+            [this.container] = jQuery('.places-edit-dialog');
```

Comment:
You can get a reference to the popup content from the popup. No need to search the whole document for it and possibly hitting the wrong element:
```suggestion
            [this.container] = this,dialog.getJqueryContent().find('.places-edit-dialog');
```

Also I'm not sure if it's a good idea to store another reference to the element here. If it's not required to have the container as a variable for other purposes you could just send the reference as a parameter for `this._renderForm(´)`

Reply:
Got rid of `container` variable and passed it straight to renderForm method.